islandora_transforms/BPL_MODS_to_solr.xslt

```diff
@@ -0,0 +1,146 @@
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Lump faceted subject topic fields together with non-date-typed names from
+       above. -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:subject/mods:topic">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'faceted_subjects'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'all_subject_descendants'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Lump together all descendants of subject for advanced search -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:subject//*">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'all_subject_descendants'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Non-otherTyped relatedItem/titleInfo/title -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:relatedItem[not(@otherType)]/mods:titleInfo/mods:title">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'relatedItem_not_otherType_titleInfo_title'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Non-typed note -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:note[not(@type)]">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'note_no_type'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Both types of call numbers, for advanced search -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:identifier[@type='local-call-number'] | mods:identifier[@type='non-marc-call-number']">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'all_call_numbers'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Both types of filing suffixes, for advanced search -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:identifier[@type='local-filing-suffix'] | mods:identifier[@type='non-marc-filing-suffix']">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'all_filing_suffixes'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Concatenate title/nonSort with title -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:titleInfo[mods:nonSort and mods:title]">
```

Comment:
Talked with Stephen when `mods:nonSort` is not present the field should still contain `mods:title`